explainer.py

```diff
@@ -0,0 +1,164 @@
+        #     print(f"Saved explanation to: {output_file}")
+        #
+        #     update_status(file_path)
+
+        time.sleep(SLEEP_TIME)
+
+
+def find_unprocessed_files():
+    """
+    Finds unprocessed files in the uploads folder.
+
+    Returns:
+        List[str]: List of file paths of unprocessed files.
+    """
+    status = load_status()
+    processed_files = set(status.keys())
+    unprocessed_files = []
+
+    for file_name in os.listdir(UPLOADS_FOLDER):
+        if file_name not in processed_files:
+            unprocessed_files.append(os.path.join(UPLOADS_FOLDER, file_name))
+
+    return unprocessed_files
+
+
+def load_status():
+    """
+    Loads the status file.
+
+    Returns:
+        dict: Dictionary containing the status information.
+    """
+    if not os.path.exists(STATUS_FILE):
+        return {}
+
+    with open(STATUS_FILE) as f:
+        status = json.load(f)
+
+    return status
+
+
+def update_status(file_path):
+    """
+    Updates the status file with the processed file.
+
+    Args:
+        file_path (str): The path of the processed file.
+    """
+    status = load_status()
+    file_name = os.path.basename(file_path)
+    status[file_name] = int(time.time())
+
+    with open(STATUS_FILE, 'w') as f:
+        json.dump(status, f)
+
+
+def load_presentation(file_path):
+    """
+    Loads a PowerPoint presentation from a file.
+
+    Args:
+        file_path (str): The path of the PowerPoint presentation file.
+
+    Returns:
+        Presentation_parser: The loaded presentation parsed object.
+    """
+    presentation = PresentationParser(file_path)
+    presentation.parse()
+    return presentation
+
+
+def generate_explanation(presentation, main_topic: str = None):
+    """
+    Generates an explanation for a PowerPoint presentation.
+
+    Args:
+        main_topic: (Str): The topic of this presentation.
+        presentation (Presentation_parser): The presentation object to generate the explanation for.
+
+    Returns:
+        dict: The generated explanation.
+    """
+    explainer = GPTSlideExpander()
+    if not main_topic:
+        main_topic = asyncio.run(explainer.generate_topic_for_presentation(
+            presentation.get_all_presentation_text()))
+    asyncio.run(
+        GPTSlideExpander.generate_explanation_for_presentation(explainer, presentation.presentation_text,
+                                                               main_topic)
+    )
+    return explainer.expanded_slide_explanations
+
+
+def save_explanation(explanation, file_path):
+    """
+    Saves the explanation to a file.
+
+    Args:
+        explanation (dict): The explanation to be saved.
+        file_path (str): The path of the original file.
+
+    Returns:
+        str: The path of the saved explanation file.
+    """
+    file_name = os.path.basename(file_path)
+    output_file = os.path.join(OUTPUTS_FOLDER, f"{file_name}.json")
+
+    with open(output_file, 'w') as f:
```

Comment:
This name does not adhere to python's naming conventions. Names should be indicative, and when used properly can reduce redundant documentation. In addition, each Variable, function, module, constant and class names has a specific naming convention. Read this link to learn the proper conventions - https://visualgit.readthedocs.io/en/latest/pages/naming_convention.html. If you don't know what CamelCase and snake_case means, read this link - https://www.google.com/amp/s/www.theserverside.com/answer/Camel-case-vs-snake-case-Whats-the-difference%3famp=1 . For a general overview on python conventions (PEP-8), read this link - https://peps.python.org/pep-0008